core/trino-main/src/test/java/io/trino/cost/TestFilterStatsCalculator.java

```diff
@@ -301,6 +311,99 @@ public void testAndStats()
 
         assertExpression("CAST(NULL AS boolean) AND CAST(NULL AS boolean)").equalTo(zeroStatistics);
         assertExpression("CAST(NULL AS boolean) AND (x < 0e0 AND x > 1e0)").equalTo(zeroStatistics);
+
+        Consumer<SymbolStatsAssertion> symbolAssertX = symbolAssert -> symbolAssert.averageRowSize(4.0)
+                .lowValue(-5.0)
+                .highValue(5.0)
+                .distinctValuesCount(20.0)
+                .nullsFraction(0.0);
+        Consumer<SymbolStatsAssertion> symbolAssertY = symbolAssert -> symbolAssert.averageRowSize(4.0)
+                .lowValue(1.0)
+                .highValue(5.0)
+                .distinctValuesCount(16.0)
+                .nullsFraction(0.0);
+
+        assertExpression(
+                "(x BETWEEN -5 AND 5) AND y > 1",
+                Session.builder(session).setSystemProperty(FILTER_CONJUNCTION_INDEPENDENCE_FACTOR, "0").build())
+                .outputRowsCount(375)
+                .symbolStats("x", symbolAssertX)
+                .symbolStats("y", symbolAssertY);
+
+        assertExpression(
+                "(x BETWEEN -5 AND 5) AND y > 1",
+                Session.builder(session).setSystemProperty(FILTER_CONJUNCTION_INDEPENDENCE_FACTOR, "1").build())
+                .outputRowsCount(150)
```

Comment:
add a comment. Is that selectivity of `(x BETWEEN -5 AND 5)` and `y > 1`?

Maybe you could extract sensitivities of these predicates as constants and use it in code, e.g:
```
.outputRowsCount(totalRowCount * X_SELECTIVITY * Y_SELECTIVITY)
```